src/study_api.jl

```diff
@@ -1,4 +1,75 @@
-const PSRCLASSES_DEFAULTS = JSON.parsefile(joinpath(@__DIR__, "JSON", "psrclasses.defaults.json"))
+const PSRCLASSES_DEFAULTS = JSON.parsefile(joinpath(@__DIR__, "json_metadata", "psrclasses.defaults.json"))
+const DATE_FORMAT_1 = Dates.DateFormat(raw"yyyy-mm-dd")
+const DATE_FORMAT_2 = Dates.DateFormat(raw"dd/mm/yyyy")
+
+function list_attributes(
+    data::Data,
+    name::String,
+)
+    if !haskey(data.data_struct, name)
+        error("PSR Class '$name' is not available for this study")
+    end
+
+    class_struct = data.data_struct[name]
+
+    attrs = sort(collect(keys(class_struct)))
+
+    return attrs
+end
+
+function list_attributes(
+    data::Data,
+    name::String,
+    index::Integer,
+)
+    element = _get_element(data, name, index)
+
+    attrs = sort(collect(keys(element)))
+
+    return attrs
+end
+
+function list_indexed_attributes(
+    data::Data,
+    name::String,
+    index_attr::String,
+)
+    if !haskey(data.data_struct, name)
+        error("PSR Class '$name' is not available for this study")
+    end
+
+    class_struct = data.data_struct[name]
+
+    attrs = [
+        attr for (attr, attr_data) in class_struct
+        if attr_data.index == index_attr || attr == index_attr
+    ] |> sort
```

Comment:
Nice list comprehension but I think that an explicit approach to this one would be better, allocate a vector make the loop with the conditional and then sort in 3 steps.